fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateTableInfo.java

```diff
@@ -525,11 +525,16 @@ public void validate(ConnectContext ctx) {
         }
     }
 
+    /**
+     * validate ctas definition
+     */
     public void validateCreateTableAsSelect(List<ColumnDefinition> columns, ConnectContext ctx) {
         this.columns = Utils.copyRequiredMutableList(columns);
         // bucket num is hard coded 10 to be consistent with legacy planner
-        this.distribution = new DistributionDescriptor(true, false, 10,
-                Lists.newArrayList(columns.get(0).getName()));
+        if (engineName.equals("olap") && this.distribution == null) {
```

Comment:
We need check it strictly.
If engine name is not OLAP and distribution is not null, should throw exception.